src/pages/DealPreparation.tsx

```diff
@@ -96,8 +94,26 @@ export function DealPreparation() {
         throw new Error("Selected provider does not exist");
       }
 
+      const collatorMaddrs = await PolkaCollatorRpc.getP2PMultiaddrs(wsAddress);
+      const wsMaddrs = collatorMaddrs
+        .filter((maddr) => maddr.includes("ws"))
```

Comment:
WS should be a constant defined somewhere

Reply:
I see this getting changed, it's kind of an hack, so I'm not sure if it is worth it